instance-scheduler/environments.go

```diff
@@ -0,0 +1,142 @@
+
+// GitHubFile represents a single file in a GitHub directory as returned by the GitHub API
+type GitHubFile struct {
+    Name string `json:"name"`
+    Path string `json:"path"`
+    Type string `json:"type"` // "file" or "dir"
+}
+
+// JSONFileContent represents the structure of each JSON file's content
+type JSONFileContent map[string]interface{}
+
+// FetchJSON fetches the JSON content from a given URL
+func FetchJSON(rawURL string) (JSONFileContent, error) {
+    resp, err := http.Get(rawURL)
+    if err != nil {
+        return nil, fmt.Errorf("failed to fetch JSON content: %w", err)
+    }
+    defer resp.Body.Close()
+
+    if resp.StatusCode != http.StatusOK {
+        return nil, fmt.Errorf("non-200 status code: %d", resp.StatusCode)
+    }
+
+    body, err := ioutil.ReadAll(resp.Body)
+    if err != nil {
+        return nil, fmt.Errorf("failed to read response body: %w", err)
+    }
+
+    var content JSONFileContent
+    if err := json.Unmarshal(body, &content); err != nil {
+        return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
+    }
+
+    return content, nil
+}
+
+// fetches the environments JSON data from GitHub
+func fetchGitHubData(repoOwner, repoName, branch, directory string) ([]byte, error) {
+    baseURL := "https://api.github.com/repos"
+    u, err := url.Parse(baseURL)
+    if err != nil {
+        return nil, fmt.Errorf("failed to parse base URL: %w", err)
+    }
+
+    u.Path = strings.Join([]string{u.Path, repoOwner, repoName, "contents", directory}, "/")
+    query := u.Query()
+    query.Set("ref", branch)
+    u.RawQuery = query.Encode()
+
+    // Print the constructed URL for debugging
+    fmt.Println("Constructed URL:", u.String())
+
+    // Create a new HTTP request
+    req, err := http.NewRequest("GET", u.String(), nil)
+    if err != nil {
+        return nil, fmt.Errorf("failed to create HTTP request: %w", err)
+    }
+
+    // Set the User-Agent header (GitHub API requires a User-Agent header)
+    req.Header.Set("User-Agent", "Go-http-client")
+
+    // Perform the HTTP request
+    client := &http.Client{}
+    resp, err := client.Do(req)
+    if err != nil {
+        return nil, fmt.Errorf("failed to fetch directory listing: %w", err)
+    }
+    defer resp.Body.Close()
+
+    if resp.StatusCode != http.StatusOK {
+        return nil, fmt.Errorf("non-200 status code: %d", resp.StatusCode)
+    }
+
+    body, err := ioutil.ReadAll(resp.Body)
+    if err != nil {
+        return nil, fmt.Errorf("failed to read response body: %w", err)
+    }
+
+    return body, nil
+}
+
+// processGitHubData processes the JSON data and returns a slice of GitHubFile
+func processGitHubData(body []byte) ([]GitHubFile, error) {
+    var files []GitHubFile
+    if err := json.Unmarshal(body, &files); err != nil {
+        return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
+    }
```

Comment:
Recommend aligning the Unmarshalls error checks like we do everywhere else - it keeps things consistent and means the reader needs to know one less go idiom to get by 🙈

```suggestion
    err = json.Unmarshal(body, &files)
    if err != nil {
        return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
    }
```